networkx/generators/geometric.py

```diff
@@ -784,3 +785,110 @@ def thresholded_random_geometric_graph(
     )
     G.add_edges_from(edges)
     return G
+
+
+@py_random_state(5)
+def S1_graph(n, beta, gamma, mean_degree, kappas=None, seed=None):
```

Comment:
```suggestion
def S1_graph(n, * ,  beta=None, gamma=None, mean_degree=None, kappas=None, seed=None):
```

We shouldn't force users to provide `beta`, `gamma`, `mean_degree` when they are already providing `kappas`. This signature will also force users to be explicit about the parameters they are passing. So they would need to call this function with something like `nx.S1_graph(100, beta=1.5, gamma=2.7, mean_degree=5)` or `nx.S1_graph(100, kappas=kappas_dict)`.